FWCore/Framework/src/InputSource.cc

```diff
@@ -84,12 +83,9 @@ namespace edm {
     }
     if (maxSecondsUntilRampdown_ > 0) {
       processingStart_ = std::chrono::steady_clock::now();
-  }
-
-    // Secondary input sources currently do not have a product registry.
-    if(primary_) {
-      assert(desc.productRegistry_ != 0);
     }
+
+    assert(desc.productRegistry_ != 0);
```

Comment:
Actually, this assert was already obsolete, because, contrary to the removed comment, secondary input sources also have product registries, and, as a result, the productRegistry_ data member of InputSourceDescription  was initialized to point to an actual product registry, so the pointer could never be 0.  The long since obsolete assert will be removed.